### PR TITLE
support for Base64.getDecoder().decode(String)

### PR DESCRIPTION
**Pull request by jgaleotti**

### CODE REVIEW

File: client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/classes/Base64DecoderClassReplacement.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.Objects;
+
+public class Base64DecoderClassReplacement implements MethodReplacementClass {
+
+    @Override
+    public Class<?> getTargetClass() {
+        return Base64.Decoder.class;
+    }
+
+    private final static String BASE64_REGEX = "^[A-Za-z0-9+/]+={0,2}$";
```
Comment (arcuri82) [inline client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/classes/Base64DecoderClassReplacement.java:21]: if this is coming from somewhere, best to put a link here as well

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: client-java/instrumentation/src/test/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/classes/Base64DecoderClassReplacementTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.evomaster.client.java.instrumentation.shared.ObjectiveNaming;
+import org.evomaster.client.java.instrumentation.staticstate.ExecutionTracer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Base64;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class Base64DecoderClassReplacementTest {
```
[the remark this answers is not in the document]
Comment (jgaleotti) [reply client-java/instrumentation/src/test/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/classes/Base64DecoderClassReplacementTest.java:13]: You're right. I was just checking with EMB why the error message kept happening. 
I will add the e2e test too.
Thxs!

File: client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/classes/Base64DecoderClassReplacement.java (hunk 1 above)
Comment (jgaleotti) [reply client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/classes/Base64DecoderClassReplacement.java:21]: I found a better regex for this. I am also adding the link to the source.

File: client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/ReplacementList.java [hunk 3]
```diff
@@ -68,7 +68,8 @@ public static List<MethodReplacementClass> getList() {
                     new URLClassReplacement(),
                     new UUIDClassReplacement(),
                     new ValidatorClassReplacement(),
-                    new WebRequestClassReplacement()
+                    new WebRequestClassReplacement(),
+                    new Base64DecoderClassReplacement()
```
Comment (arcuri82) [inline client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/ReplacementList.java:72]: list should be kept in alphabetic order

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/ReplacementList.java (hunk 3 above)
Comment (jgaleotti) [reply client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/ReplacementList.java:72]: ok

**Review by arcuri82 — APPROVED** [review]

**Pull request merged**